storage/storage-interface/src/cached_state_view.rs

```diff
@@ -214,14 +219,14 @@ impl StateView for CachedStateView {
 
 pub struct CachedDbStateView {
     db_state_view: DbStateView,
-    state_cache: RwLock<HashMap<StateKey, Option<Vec<u8>>>>,
+    state_cache: DashMap<StateKey, Option<Vec<u8>>>,
 }
 
 impl From<DbStateView> for CachedDbStateView {
     fn from(db_state_view: DbStateView) -> Self {
         Self {
             db_state_view,
-            state_cache: RwLock::new(HashMap::new()),
+            state_cache: DashMap::new(),
```

Comment:
Have you done benchmarks on this?  I know this is a commonly used pattern across our code base, and we had removed Dashmap back at Diem.  I wasn't sure if it was because of performance reasons or in search of removing dependencies.

If it is faster, we should look into what kind of performance improvements it could use elsewhere.

Also, keeping in mind hashmaps make determinism harder for testing.

Reply:
@gregnazario - Yes we are working on benchmarking this. https://github.com/aptos-labs/aptos-core/pull/5791 is the relevant PR that adds counters around cache read/write latency.

Reply:
Hey @gregnazario yes, we are doing this because we are suspecting a contention bottleneck here (added you to the discussion).
Won't land until we confirm performance improvement, of course.
Generally speaking, RwLocks are terrible even just for readers beyond certain concurrency, and RwLock<HashMap pattern shouldn't be used in these scenarios, which is exactly the scenario DashMap is designed for (e.g. even in the simple scheme, which I believe is used to implement it, RwLock on hash buckets/shards reduces number of readers / contention per shard). Agreed should methodically look elsewhere as well.

Why would determinism change between hashmap and dashmap?

Reply:
Let's verify this for different scenarios. (low load v.s. high load, small db v.s. large db, etc.)